### PR TITLE
SwiftUI Template

### PR DESCRIPTION
**Pull request by fortmarek**
Resolves https://github.com/tuist/tuist/issues/450

### Short description 📝

This PR leverages a new `scaffold` feature - now you will be able to run `tuist init --template swiftui` and it will initialize application that is ready for SwiftIU!

### Solution 📦

Instead initializing `UIViewController` we initialize `UIHostingController` with `ContentView`. Besides that, I took the liberty to add `SceneDelegate` as that is a default way for new apps that are iOS 13.0+ (which SwiftUI apps have to be). (which also necessitated some changes in `InfoPlist`)

### Implementation 👩‍💻👨‍💻

I have had to refactor the default template and extracted the shared `.stencil` files right into `Templates` dir instead of `Templates/default`. You can see that the new SwiftUI `Template.swift` manifest is a little bit copy&paste -> this could normally be fixed by `ProjectDescriptionHelpers`, but as we have nothing like that in Tuist-provided templates, I feel this is an OK solution.

- [x] Refactor `default`'s `Template.swift`
- [x] Add swiftui template
- [x] Add fixture tests
- [ ] Add changelog

### CODE REVIEW

**Comment by fortmarek** [comment]
I see that the codecov is failing, but I have added almost no business logic (and for the little I have, I have added a test), so I feel that it'd be better to ignore it this time.

File: Sources/TuistKit/Commands/InitCommand.swift [hunk 1]
```diff
@@ -194,27 +196,35 @@ class InitCommand: NSObject, Command {
     /// - Returns: Array of parsed attributes
     private func validateAttributes(_ attributes: [String: OptionArgument<String>],
                                     template: Template,
+                                    name: String,
+                                    platform: Platform,
                                     arguments: ArgumentParser.Result) throws -> [String: String] {
-        try template.attributes.reduce([:]) {
-            var mutableDict = $0
+        try template.attributes.reduce(into: [:]) {
+            if $1.name == "name" {
```
Comment (pepibumur) [inline Sources/TuistKit/Commands/InitCommand.swift:203]: What about naming $0 and $1? Considering this is not a one-line closure, it makes it easier to read the code.

File: Sources/TuistLoader/Loaders/TemplateLoader.swift [hunk 2]
```diff
@@ -27,17 +27,18 @@ public class TemplateLoader: TemplateLoading {
 
     public func loadTemplate(at path: AbsolutePath) throws -> TuistCore.Template {
         let template = try manifestLoader.loadTemplate(at: path)
+        let generatorPaths = GeneratorPaths(manifestDirectory: path)
         return try TuistCore.Template.from(manifest: template,
-                                           at: path)
+                                           generatorPaths: generatorPaths)
```
Comment (pepibumur) [inline Sources/TuistLoader/Loaders/TemplateLoader.swift:32]: Nice! So we can use now relative paths 🎉

File: Sources/TuistScaffold/TemplateGenerator.swift [hunk 3]
```diff
@@ -81,7 +81,7 @@ public final class TemplateGenerator: TemplateGenerating {
                     renderedContents = fileContents
                 }
             }
-
+            guard !renderedContents.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else { return }
```
Comment (pepibumur) [inline Sources/TuistScaffold/TemplateGenerator.swift:84]: Why is this necessary? Should we add a comment?

File: Templates/SupportTests.stencil [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import Foundation
```
Comment (pepibumur) [inline Templates/SupportTests.stencil:1]: What about giving the `Templates` directory a structure that represents the templates?
```
Templates/
  default/
  swiftui/
```
Rather than placing all the files under `Templates/`? That way one can spot which files belong to which template.

File: features/step_definitions/init.rb [hunk 5]
```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
-When(/I initialize a (.+) application named (.+)/) do |platform, name|
+When(/^I initialize a (.+) application named (.+) with (.+) template$/) do |platform, name, template|
```
Comment (pepibumur) [inline features/step_definitions/init.rb:3]: ❤️ You are becoming an acceptance tests expert!

File: Sources/TuistScaffold/TemplateGenerator.swift (hunk 3 above)
Comment (fortmarek) [reply Sources/TuistScaffold/TemplateGenerator.swift:84]: Yeah, I'll add one -> if a file has no characters, we make the assumption that the user did not want to generate this file.
This is then leveraged in some files that are macOS or iOS-only.

File: Templates/SupportTests.stencil (hunk 4 above)
Comment (fortmarek) [reply Templates/SupportTests.stencil:1]: That's how it works. Files under `Templates` are shared between `default` and `swiftui`

Comment (pepibumur) [reply Templates/SupportTests.stencil:1]: What about:

```
Templates/
  shared/
  default/
  swiftui/
```

Comment (fortmarek) [reply Templates/SupportTests.stencil:1]: That will create a problem since a folder is directly translated to a template. I can modify this, but I don't like the fact that then it's a little bit more ambiguous what's a template and what's not (you will not be able to see that just by looking at that directory). Users should put shared templates in `ProjectDescriptionHelpers`, this is just a workaround for tuist built-in ones.

File: features/step_definitions/init.rb (hunk 5 above)
Comment (fortmarek) [reply features/step_definitions/init.rb:3]: It's becoming a joy to write these!

File: Sources/TuistLoader/Loaders/TemplateLoader.swift (hunk 2 above)
Comment (fortmarek) [reply Sources/TuistLoader/Loaders/TemplateLoader.swift:32]: Yes! This should make it easier for users to define templates with shared template files.

**Comment by pepibumur** [comment]
Should we ship it? 🚢

**Comment by fortmarek** [comment]
I think so! I can't do it because of the failing pipeline, but please go ahead 🎊
